### PR TITLE
add JSON junk filter to async_wrapper

### PR DESCRIPTION
**Pull request by nitzmahone**
##### ISSUE TYPE
- Bugfix Pull Request
##### COMPONENT NAME

async_wrapper.py

This needs to be merged before https://github.com/ansible/ansible/pull/17834

### CODE REVIEW

File: utilities/logic/async_wrapper.py [hunk 1]
```diff
@@ -70,6 +70,49 @@ def daemonize_self():
     os.dup2(dev_null.fileno(), sys.stdout.fileno())
     os.dup2(dev_null.fileno(), sys.stderr.fileno())
 
+# FUTURE: cloned from ActionBase. Move this to module_utils to share between core/async_wrapper, AnsibleModule-ify this module so it's Ansizballz-compatible
```
Comment (abadger) [inline utilities/logic/async_wrapper.py]: If you follow my suggestion in https://github.com/ansible/ansible/pull/17834 to move the function out of action/**init**.py and into module_utils now, update this comment to point to the file that the code lives in.  Basically, we want to let future maintainers know that they have to make any changes in both places.

**Comment by abadger** [comment]
Looks good.  Let's make the catchall exception on line 166:

``` python
except (ValueError, Exception):
```

That way if someone comes along later and tries to clean up catchalls they'll see that we definitely anticipate ValueError and want to handle it that way.

**Comment by nitzmahone** [comment]
Good call- I actually handled ValueError explicitly in one version of this, but it worked better in the catchall.

**Comment by gregdek** [comment]
Thanks @nitzmahone for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**